### ISSUE TITLE
No error when unable to connect

### ISSUE THREAD

**Issue by mkock** [secsy/goftp#35]
I was running an FTP server in a local Docker container, MacOSX Mojave. I accidentally started my program while Docker was not running and noticed that it continued with its business although the FTP connection failed.

It turns out that `goftp.DialConfig` doesn't return an error, even though it logs the failed connection attempt.

Source:

```
	dialConf := goftp.Config{
		User:               "bob",
		Password:           "12345",
		ConnectionsPerHost: 1,
		Timeout:            12 * time.Hour,
		Logger:             os.Stderr,
	}
	fmt.Printf("Connecting to %s:%d...\n", "...", "...")
	client, dialErr := goftp.DialConfig(dialConf, "127.0.0.1:21")
	fmt.Println(client, dialErr)
	if dialErr != nil {
		return dialErr
	}
```

This outputs:

```
Connecting to 127.0.0.1:21...
&{{bob 12345 1 43200000000000 <nil> 0 false 0xc0000a0010 0x1509ae0 false :0 false map[]} [127.0.0.1:21] 0xc0000a2420 map[] map[] 0 0 {0 0} {13757273498589639176 1641114 0x150e0c0} false} <nil>
goftp: 0.000 #1 opening control connection to 127.0.0.1:21
goftp: 0.000 #1 closing
goftp: 0.000 #1 error connecting: dial tcp 127.0.0.1:21: connect: connection refused
```

Notice the `<nil>` value, which is a dump of `dialErr`, and the log output stating that the connection was refused. To fail correctly, I would probably need to run a harmless operation such as STAT on a directory before the error will appear.

**Comment by muirmanders** [comment]
Yes, well spotted. `Dial` is a misnomer since it doesn't synchronously attempt to open a connection like other Dials in in go. You won't get a connection error until you try to perform an operation.

Perhaps better documentation for `Dial` would help. I plan to do a "v2" release at some point to fix this and other issues in the public API.

**Comment by valleen** [comment]
Hi,
Is there any news on this?
thx
